File: JavaScript/7js 2/isPrime.js
/*
|--------------------------------------------------------------------------
| Check for prime numbers
|--------------------------------------------------------------------------
|
*/

function isPrime(num) {
   for (let i = 2; i < num; i++)
      if (num % i === 0) return false;
   return num > 1;
}


const isPrime = (n, divisor = 2) =>
   n < 2 ? false : divisor === n ? true : n % divisor === 0 ? false : isPrime(n, divisor + 1);


const isPrime = (n) => {
   let m = 0;
   for (let i = 1; i <= n; i++) {
      Number.isInteger(n / i) && m++;
   }
   return m === 2;
};


const isPrime = n =>
   !/^x?$|^(xx+?)\1+$/.test(`x`.repeat(n));


const isPrime = length => length > 1 ?
   Array.from({
      length
   }, (_, i) => length % ++i === 0).filter(Boolean).length === 2 : false


console.log(isPrime(0), false)
console.log(isPrime(1), false)
console.log(isPrime(2), true)
console.log(isPrime(11), true)
console.log(isPrime(12), false)
console.log(isPrime(61), true)
console.log(isPrime(571), true)
console.log(isPrime(573), false)
console.log(isPrime(25), false)

File: Python/7py 2/is_prime.py
"""
|--------------------------------------------------------------------------
| Check for prime numbers
|--------------------------------------------------------------------------
|
"""

def is_prime(n):
   for i in range(2, n):
      if n % i == 0:
         return False
   return n > 1


def is_prime(n):
   return n > 1 and all(n % i for i in range(2, int(n**0.5) + 1))  


def is_prime(n,i=2):
   if (n <= 2):
      return True if(n == 2) else False
   elif (n % i == 0):
      return False
   elif (i * i > n):
      return True
   return is_prime(n, i + 1)


def is_prime(n):
   return pow(2, n - 1, n) == 1 if n > 2 else n == 2


def is_prime(num):
   return num > 1 and not any(num % n == 0 for n in range(2, num))


print(is_prime(0),False)
print(is_prime(1),False)
print(is_prime(2),True)
print(is_prime(3),True)
print(is_prime(11),True)
print(is_prime(12),False)
print(is_prime(571),True)
print(is_prime(25),False)